sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/LockRenewalOperation.java

```diff
@@ -25,7 +25,7 @@
 /**
  * Represents a renewal session or message lock renewal operation that.
  */
-class LockRenewalOperation implements AutoCloseable {
+class LockRenewalOperation implements AutoCloseable, Disposable {
```

Comment:
Curious why we also implement Disposable if it calls a method on the AutoCloseable class.

Reply:
It was added as a convenience so that operation can be added to Disposable.composite object. For example, this V2 [code](https://github.com/Azure/azure-sdk-for-java/blob/e04a8291ca335acda3010eb95b9f4cf888361221/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionReactorReceiver.java#L53). This way could simplify & avoid an extra variable (e.g. V1 [code](https://github.com/Azure/azure-sdk-for-java/blob/1333e7093d45dc92c47f528dd2836999d77c763b/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionReceiver.java#L170)) to track the async-closeable. We also treat operation as Disposable in V2 Processor auto-lock-renew.

Reply:
Btw we are not scoping LockRenewalOperation instance using try-with anywhere. So I wonder once we are on v2 completely, if we could limit the extends only to Disposable?